### PR TITLE
feat(integrations): add qiwi donate integration

### PR DESCRIPTION
**Pull request by Banonevar**
###### CHECKLIST

- [ ] tests are changed or added
- [ ] documentation is changed or added
- [ ] locales are changed or added
- [ ] db relationship (tools/database) are changed or added
- [ ] commit message follows [commit guidelines](https://github.com/sogehige/sogeBot/blob/master/CONTRIBUTING.md#commit-guidelines)

### CODE REVIEW

File: src/bot/integrations/qiwi.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      clearInterval(this.interval);
+    }
+  }
+
+  async start () {
+    clearInterval(this.interval);
+    if (this.secretToken.trim() === '' || !this.enabled) {
+      return;
+    } else {
+      this.interval = setInterval(() => this.check(), 3000);
```
Comment (sogehige) [inline src/bot/integrations/qiwi.ts]: Isn't 3000 too fast? I think minute or half minute should be OK

File: src/bot/integrations/qiwi.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      error(`Qiwi: error on api request: ${e.message}`);
+      return;
+    }
+    const data = request.data;
+    if (data.events.length === 0) {
+      return;
+    }
+    for (const event of data.events) {
+      const { DONATION_SENDER, DONATION_AMOUNT, DONATION_CURRENCY, DONATION_MESSAGE } = event.attributes;
+      const username = DONATION_SENDER ? DONATION_SENDER : 'Anonymous';
```
Comment (sogehige) [inline src/bot/integrations/qiwi.ts]: would like to see `null` instead of Anonymous

File: src/bot/integrations/qiwi.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      const message = DONATION_MESSAGE ? DONATION_MESSAGE : '';
+      const amount = Number(DONATION_AMOUNT);
+      const currency = DONATION_CURRENCY;
+
+      const id = await global.users.getIdByName(DONATION_SENDER.toLowerCase(), false);
```
Comment (sogehige) [inline src/bot/integrations/qiwi.ts]: username null check

File: src/bot/integrations/qiwi.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        global.db.engine.insert('users.tips', { id, amount, message, currency, timestamp: Date.now() });
+      }
+      if (await global.cache.isOnline()) {
+        await global.db.engine.increment('api.current', { key: 'tips' }, { value: parseFloat(global.currency.exchange(amount, currency, global.currency.mainCurrency)) });
+      }
+      global.overlays.eventlist.add({
+        type: 'tip',
+        amount,
+        currency,
+        username,
```
Comment (sogehige) [inline src/bot/integrations/qiwi.ts]: username || 'Anonymous'

File: src/bot/integrations/qiwi.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        message,
+        timestamp: Date.now(),
+      });
+      
+      tip(`${DONATION_SENDER.toLowerCase()}${id ? '#' + id : ''}, amount: ${amount}${DONATION_CURRENCY}, ${message ? 'message: ' + message : ''}`);
```
Comment (sogehige) [inline src/bot/integrations/qiwi.ts]: use username variable, 
username + id || 'Anonymous'

File: src/bot/integrations/qiwi.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+      global.events.fire('tip', {
+        username,
```
Comment (sogehige) [inline src/bot/integrations/qiwi.ts]: username || 'Anonymous'

File: src/bot/integrations/qiwi.ts [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        amount,
+        currency,
+        amountInBotCurrency: parseFloat(global.currency.exchange(amount, currency, global.currency.mainCurrency)).toFixed(2),
+        currencyInBot: global.currency.mainCurrency,
+        message,
+      });
+
+      global.registries.alerts.trigger({
+        event: 'tips',
+        name: username,
```
Comment (sogehige) [inline src/bot/integrations/qiwi.ts]: username || 'Anonymous'

File: src/bot/integrations/qiwi.ts [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        amount,
+        currency,
+        monthsName: '',
+        message,
+        autohost: false,
+      });
+
+      triggerInterfaceOnTip({
+        username: username,
```
Comment (sogehige) [inline src/bot/integrations/qiwi.ts]: username || 'Anonymous'

File: src/bot/integrations/qiwi.ts (hunk 1 above)
Comment (Banonevar) [reply src/bot/integrations/qiwi.ts]: No, it's ok. They haven't rate limits. i asked platform support about that.

File: src/bot/integrations/qiwi.ts [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+      global.overlays.eventlist.add({
+        type: 'tip',
+        amount,
+        currency,
+        username: username || 'Anonymous',
+        message,
+        timestamp: Date.now(),
+      });
+      
+      tip(`${username}${id ? '#' + id : ''}, amount: ${amount}${DONATION_CURRENCY}, ${message ? 'message: ' + message : ''}`);
```
Comment (sogehige) [inline src/bot/integrations/qiwi.ts]: add null check (Anonymous without id) here as well. otherwise :+1:

**Pull request merged**